sycl/cmake/modules/AddSYCLUnitTest.cmake

```diff
@@ -6,7 +6,8 @@ macro(add_sycl_unittest test_dirname link_variant)
   # Enable exception handling for these unit tests
   set(LLVM_REQUIRES_EH 1)
 
-  if (MSVC AND CMAKE_BUILD_TYPE MATCHES "Debug")
+  string(TOLOWER "${CMAKE_BUILD_TYPE}" build_type_lower)
```

Comment:
I would report an error if CMAKE_BUILD_TYPE is not "Debug".

Reply:
I can't agree with it.
See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
```
Note that configuration names are case-insensitive. The value of this variable will be the same as it is specified when invoking CMake. For instance, if -DCMAKE_BUILD_TYPE=ReLeAsE is specified, then the value of CMAKE_BUILD_TYPE will be ReLeAsE.
```

Reply:
I don't your point. The documentation says that value of CMAKE_BUILD_TYPE is case-insensitive, but my point is "Let DPC++ CMake files support only one spelling version - Debug and reject all the others e.g. DeBuG".

See the same document: 
> Possible values are empty, Debug, Release, RelWithDebInfo, MinSizeRel, ...

Reply:
Still, cmake isn't sensitive to case of value of `CMAKE_BUILD_TYPE` i.e. if it's `DeBuG` or `debug` etc., then cmake acts the same as if it was `Debug`.

Reply:
If you prefer to support all spelling, could you set `build_type_lower` string once and re-use in both macro functions?

Reply:
Updated.